### DISCUSSION TITLE
Rerunning a notebook does not clean up CUDA memory

### DISCUSSION THREAD

**Discussion by Yelinz** [marimo-team/marimo#797] [Q&A] [answered]
When I load a model and change something which causes the cell with the model loading to be rerun (adding a new import).
It will complain that there is no more memory on CUDA.
Because the previous model was not unloaded, but subsequent cell cannot be run then because the previous one erred with no memory.

Calling `torch.cuda.empty_cache()` also does nothing.

Now the state of the notebook and memory is broken and I have to restart the marimo server every time it tries to rerun the model loading cell.

Any ideas on how to alleviate this?

**Comment by mscolnick** [comment, marked as answer]
this was fixed in https://github.com/marimo-team/marimo/pull/807
